qiskit_dynamics/models/lindblad_model.py

```diff
@@ -0,0 +1,616 @@
+
+        # convert static hamiltonian to new frame setup
+        static_ham = self.get_static_hamiltonian(in_frame_basis=True)
+        if static_ham is not None:
+            static_ham = -1j * static_ham
+
+        new_static_hamiltonian = GeneratorModel.transfer_static_operator_between_frames(
+            static_ham,
+            new_frame=new_frame,
+            old_frame=self.rotating_frame,
+        )
+
+        if new_static_hamiltonian is not None:
+            new_static_hamiltonian = 1j * new_static_hamiltonian
+
+        # convert hamiltonian operators and dissipator operators
+        ham_ops = self.get_hamiltonian_operators(in_frame_basis=True)
+        static_diss_ops = self.get_static_dissipators(in_frame_basis=True)
+        diss_ops = self.get_dissipator_operators(in_frame_basis=True)
+
+        new_hamiltonian_operators = GeneratorModel.transfer_operators_between_frames(
+            ham_ops,
+            new_frame=new_frame,
+            old_frame=self.rotating_frame,
+        )
+        new_static_dissipators = GeneratorModel.transfer_operators_between_frames(
+            static_diss_ops,
+            new_frame=new_frame,
+            old_frame=self.rotating_frame,
+        )
+        new_dissipator_operators = GeneratorModel.transfer_operators_between_frames(
+            diss_ops,
+            new_frame=new_frame,
+            old_frame=self.rotating_frame,
+        )
+
+        self._rotating_frame = new_frame
+
+        self._operator_collection = self.construct_operator_collection(
+            evaluation_mode=self.evaluation_mode,
+            static_hamiltonian=new_static_hamiltonian,
+            hamiltonian_operators=new_hamiltonian_operators,
+            static_dissipators=new_static_dissipators,
+            dissipator_operators=new_dissipator_operators,
+        )
+
+    def evaluate_hamiltonian(self, time: float, in_frame_basis: Optional[bool] = False) -> Array:
+        """Evaluates Hamiltonian matrix at a given time.
+
+        Args:
+            time: The time at which to evaluate the hamiltonian.
+            in_frame_basis: Whether to evaluate in the basis in which
+                the frame operator is diagonal.
+        Returns:
+            Array: Hamiltonian matrix."""
+
+        hamiltonian_sig_vals = None
+        if self._hamiltonian_signals is not None:
+            hamiltonian_sig_vals = self._hamiltonian_signals(time)
+
+        ham = self._operator_collection.evaluate_hamiltonian(hamiltonian_sig_vals)
+        if self.rotating_frame.frame_diag is not None:
+            ham = self.rotating_frame.operator_into_frame(
+                time,
+                ham,
+                operator_in_frame_basis=True,
+                return_in_frame_basis=in_frame_basis,
+                vectorized_operators=self.vectorized_operators,
+            )
+
+        return ham
+
+    def evaluate(self, time: float, in_frame_basis: Optional[bool] = False) -> Array:
```

Comment:
If `in_frame_basis` was changed to property all the evaluation methods / call methods could be changed to look that up too.

Reply:
Good idea.